core/src/main/java/uk/gov/gchq/koryphe/impl/predicate/IsShorterThan.java

```diff
@@ -77,20 +77,9 @@ public boolean test(final Object input) {
     }
 
     private int getLength(final Object value) {
-        final int length;
-        if (value instanceof String) {
-            length = ((String) value).length();
-        } else if (value instanceof Object[]) {
-            length = ((Object[]) value).length;
-        } else if (value instanceof Collection) {
-            length = ((Collection) value).size();
-        } else if (value instanceof Map) {
-            length = ((Map) value).size();
-        } else {
-            throw new IllegalArgumentException("Could not determine the size of the provided value");
-        }
+        final Length delegate = new Length();
```

Comment:
You could store this delete object as a field, so we don't have to recreate it every time getLength is called.